src/latexify/codegen/function_codegen.py

```diff
@@ -363,6 +366,35 @@ def visit_Call(self, node: ast.Call) -> str:
         arg_strs = [self.visit(arg) for arg in node.args]
         return lstr + ", ".join(arg_strs) + rstr
 
+    def _generate_matrix(self, node: ast.Call) -> str:
+        """Generates a matrix from numpy.matrix."""
+        if len(node.args) != 1:
+            # raise with function name
+            raise ValueError(
+                f"Invalid number of arguments {len(node.args)}"
+            )
+        
+        arg = node.args[0]
+        # TODO: Support string being passed to numpy.matrix.
+        if not isinstance(arg, ast.List):
+            raise exceptions.LatexifyNotSupportedError(
+                "Only matrices represented by lists are supported."
+            )
+
+        # TODO: Support other types of matrix environments like (pmatrix, matrix, Bmatrix, etc.)
+        latex = r"\begin{bmatrix}"
+        if type(arg) == ast.List:
+            for elt in arg.elts:
+                if type(elt) == ast.List:
+                    for sub_elt in elt.elts:
+                        latex += self.visit(sub_elt) + r" & "
+                    latex = latex[:-2] + r" \\ "
+                else:
+                    latex += self.visit(elt) + r" & "
+            latex = latex[:-3] + r"\end{bmatrix}"
```

Comment:
I think the following steps could be more intuitive and much efficient:

1. Checks if the arg is List, and all elts are List with the same length. Stops the process if the condition is not satisfied.
2. ```python
   # This should work without any breakages iff the condition 1 is satisfied.
   contents = r" \\ ".join(
       " & ".join(self.visit(x) for x in row.elts)
       for row in arg.elts
   )
   return r"\begin{bmatrix} " + contents + r" \end{bmatrix}"
   ```

Reply:
I was thinking something like this to properly handle row, column and mxn matrices correctly. What do you think?

```python
        if isinstance(arg, ast.List):
            # handle row vectors
            if all(isinstance(x, ast.Constant) for x in arg.elts):
                contents = " & ".join([self.visit(x) for x in arg.elts])
                return r"\begin{bmatrix} " + contents + r" \end{bmatrix}"
            # handle column vector
            if all(isinstance(x, ast.List) for x in arg.elts) and all(len(x.elts) == 1 for x in arg.elts):
                contents = r" \\ ".join([self.visit(x.elts[0]) for x in arg.elts])
                return r"\begin{bmatrix} " + contents + r" \end{bmatrix}"
            # handle matrices
            elif all(isinstance(elt, ast.List) and len(elt.elts) == len(arg.elts) for elt in arg.elts):
                contents = r" \\ ".join(
                    " & ".join(self.visit(x) for x in row.elts)
                    for row in arg.elts
                )
                return r"\begin{bmatrix} " + contents + r" \end{bmatrix}"
        return None
```

Reply:
It looks still wrong I think:

- 1st condition looks too strict because it introduced another constraint that the member is only `Constant`.
- 2nd condition is not necessary if we handle $M \times N$ matrix.
- 3rd condition looks to handle only square matrices.

I tried to complete it, and noticed that it requrie more detailed checks. How about it:

<img width="566" alt="無題" src="https://user-images.githubusercontent.com/1023695/202424226-05638e85-4ef5-49ce-9220-cf271cb73e3a.png">

Reply:
(btw, I think the above examples can be used in unit tests as well.)